### PR TITLE
Primed solving of https://github.com/ornicar/php-github-api/issues/13, implementation of the PullRequest API.

### PR DESCRIPTION
**Pull request by jeanvoye**
Primed solving of https://github.com/ornicar/php-github-api/issues/13,
based on Github's Pull Request API documentation : http://develop.github.com/p/pulls.html

A bunch of TODOs left, namely regarding return values from Github's API, which should probably be handled through exceptions. I could not see much analysis of the return values, did not dare taking a revamping initiative, but will gladly bring additions if you think this makes sense.

Thanks for this nice API,
Nicolas

### CODE REVIEW

File: lib/Github/Api/PullRequest.php [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+     * @param   int $issueId              If a pull-request is related to an issue, place issue ID here. The $title-$body pair and this are mutually exclusive.
+     * @return  array                     array of pull requests for the project
+     */
+    public function create($username, $repo, $base, $head, $title = null, $body = null, $issueId = null)
+    {
+        $postParameters = array( 'pull[base]' => $base,
+                                 'pull[head]' => $head
+                          );
+
+        if ( $title !== null and $body !== null )
```
Comment (ornicar) [inline lib/Github/Api/PullRequest.php]: Coding standards: the bracket opens on the same line.

File: lib/Github/Api/PullRequest.php [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        {
+            $postParameters = array_merge( $postParameters,
+                                           array(
+                                             'pull[title]' => $title,
+                                             'pull[body]'  => $body
+                                           )
+                                         );
+        }
```
Comment (ornicar) [inline lib/Github/Api/PullRequest.php]: CS:

```
if () {
} elseif () {
} else {
}
```

File: lib/Github/Api/PullRequest.php [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            $postParameters = array_merge( $postParameters,
+                                           array(
+                                             'pull[issue]' => $issueId
+                                           )
+                                         );
+        }
+        else
+        {
+            // @FIXME : Exception required here.
+            return null;
```
Comment (ornicar) [inline lib/Github/Api/PullRequest.php:79]: I think we can have a `lib/Github/Exception` dir with `Github_Exception extends Exception` class, and maybe more precise exception classes. What do you think?

File: test/Github/Tests/Api/PullRequestTest.php [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                  );
+
+        $api->create('ezsystems', 'ezpublish', 'master', 'virtualtestbranch', null, null, 25 );
+    }
+
+    protected function getApiClass()
+    {
+        return 'Github_Api_PullRequest';
+    }
+}
```
Comment (ornicar) [inline test/Github/Tests/Api/PullRequestTest.php:84]: Oh yeah, I love tested pull requests :)

**Comment by ornicar** [comment]
Great stuff ! Care to write the corresponding doc?

**Comment by nfrp** [comment]
Updated CS + updated doc + an addition forgotten in the initial PR are now part of this pull request.
Regarding the Exceptions discussion : probably something that should be added at the heart of this library, not only in specific places. How about opening a separate issue for this topic, where discussion can safely occur ?

Cheers,

**Comment by nfrp** [comment]
I realize i pushed the last comment with my professional github account. But this is @jeanvoye too :)

**Comment by ornicar** [comment]
Great job, thanks!

**Pull request merged**